### PR TITLE
feat(app): flow editor autosave

### PR DESCRIPTION
**Pull request by VojtechVidra**

### CODE REVIEW

File: apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx [hunk 1]
```diff
@@ -90,20 +84,10 @@ export const FlowEditForm: FC<Props> = ({ flow, organizationId }) => {
...
-              >
-                Save and close
-              </Button>
-              <FlowPublishChangesDialog
-                isDirty={formState.isDirty}
-                flow={flow}
-                onSave={handleSave}
-              />
+              <FlowPublishChangesDialog flow={flow} />
               <Button variant="ghost" size="icon" asChild>
```
Comment (OPesicka) [inline apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx:91]: What about disabling the close until the changes are saved? Or having some general "You have unsaved changes" dialog?

**Review by OPesicka — CHANGES_REQUESTED** [review]

File: apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/remove-draft.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      <DialogContent>
+        <Text>Are you sure you want to discard the draft? This action cannot be undone.</Text>
+      </DialogContent>
+      <DialogActions>
+        <DialogClose asChild>
+          <Button shadow="none" size="small" variant="secondary">
+            Close
+          </Button>
+        </DialogClose>
+        {/* TODO: make this close on click */}
```
Comment (OPesicka) [inline apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/remove-draft.tsx:49]: @VojtechVidra I couldn't figure this out 😄

File: apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx (hunk 1 above)
Comment (VojtechVidra) [reply apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx:117]: I've added two changes:
- the page will prompt you beforeunload (refresh or close) and save so you can actually cmd + w, esc, cmd + w
- the form will save in the background if you navigate to other page within app

File: apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/remove-draft.tsx (hunk 2 above)
Comment (VojtechVidra) [reply apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/remove-draft.tsx:53]: Fixed 👍 😄

File: apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx (hunk 1 above)
Comment (OPesicka) [reply apps/app/src/app/(dashboard)/org/[organizationId]/project/[projectId]/flow/[flowId]/edit/flow-edit-form.tsx:117]: Nice!

**Pull request merged**